components/EmotionList.js

```diff
@@ -9,40 +9,107 @@ export default function EmotionList({
   myBookmarkedEmotions,
 }) {
   const [selectedFilter, setSelectedFilter] = useState("");
+  const [showLeftArrow, setShowLeftArrow] = useState(false);
+  const [showRightArrow, setShowRightArrow] = useState(false);
+  const scrollContainerRef = useRef(null);
 
   const filteredEmotions = selectedFilter
     ? emotions.filter((emotion) => emotion.emotionType === selectedFilter)
     : emotions;
 
+  function updateArrowVisibility() {
+    const tabsBox = scrollContainerRef.current;
+    if (tabsBox) {
+      const canScrollLeft = tabsBox.scrollLeft > 0;
+      const canScrollRight =
+        tabsBox.scrollWidth > tabsBox.clientWidth + tabsBox.scrollLeft;
+      setShowLeftArrow(canScrollLeft);
+      setShowRightArrow(canScrollRight);
+    }
+  }
+
+  function handleScroll() {
+    updateArrowVisibility();
+  }
```

Comment:
instead onf handleScroll you can pass updateArrowVisibility to the eventHandler prop.